packages/valory/skills/liquidity_trader_abci/behaviours/fetch_strategies.py

```diff
@@ -3783,3 +3795,75 @@ def _track_eth_transfers_mode(
         except Exception as e:
             self.context.logger.error(f"Error tracking Mode ETH transfers: {e}")
             return {"incoming": {}, "outgoing": {}}
+
+    def get_master_safe_address(self) -> Generator[None, None, Optional[str]]:
+        """
+        Get the master safe address by checking service staking state.
+
+        Returns:
+            The master safe address if found, None otherwise.
+        """
+        # Get service_id from params
+        service_id = self.params.on_chain_service_id
+        if service_id is None:
+            self.context.logger.error("No service ID configured")
+            return None
+
+        if not self.params.target_investment_chains:
+            self.context.logger.error("No target investment chains configured")
+            return None
+
+        operating_chain = self.params.target_investment_chains[0]
+
+        staking_token_address = self.params.staking_token_contract_address
+        staking_chain = self.params.staking_chain
+
+        if staking_token_address and staking_chain:
+            if self.service_staking_state == StakingState.UNSTAKED:
+                yield from self._get_service_staking_state(chain=staking_chain)
+
+            if self.service_staking_state != StakingState.UNSTAKED:
+                service_info = yield from self._get_service_info(staking_chain)
+                if service_info and len(service_info) >= 2:
+                    master_safe_address = service_info[
+                        1
+                    ]  # owner field from service info struct
+                    self.context.logger.info(
+                        f"Master safe address: {master_safe_address}"
+                    )
+                    return master_safe_address
+                else:
+                    self.context.logger.error(
+                        "Failed to get service info from staking contract"
+                    )
+                    return None
+
+        service_registry_address = self.params.service_registry_contract_addresses.get(
+            operating_chain
+        )
+        if not service_registry_address:
+            self.context.logger.error(
+                f"No service registry address configured for operating chain {operating_chain}"
+            )
+            return None
+
+        # Get service owner from service registry
+        service_owner_result = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,
+            contract_address=service_registry_address,
+            contract_public_id=ServiceRegistryContract.contract_id,
+            contract_callable="get_service_owner",
```

Comment:
@DIvyaNautiyal07 we need to fix this with ownerOf(service_id)[https://gnosisscan.io/address/0x9338b5153AE39BB89f50468E608eD9d764B755fD#readContract#F26] method as mentioned by Andrey